packages/runner/src/builtins/map.ts

```diff
@@ -57,13 +57,23 @@ export function map(
   // between the source list and the result list.
   let initializedUpTo = 0;
 
+  // Keep track of result cells so they can be cancelled when the list shrinks
+  const resultCells: DocImpl<any>[] = [];
+  const resultCancels: Array<(() => void) | undefined> = [];
```

Comment:
is this a good idea?  should this actually be using a cancelGroup and returning it.